rebac-admin-backend/v1/response.go

```diff
@@ -56,10 +58,35 @@ func writeResponse(w http.ResponseWriter, status int, responseObject interface{}
 		return
 	}
 
+	setJSONContentTypeHeader(w)
 	w.WriteHeader(status)
+	w.Write(data)
+}
 
-	_, err = w.Write(data)
-	if err != nil {
-		writeErrorResponse(w, err)
+// mapServiceErrorResponse maps errors thrown by services to the designated
+// response type. If the given mapper is nil, the method uses the default
+// mapping strategy.
+//
+// This method should never return nil response.
+func mapServiceErrorResponse(mapper ErrorResponseMapper, err error) *resources.Response {
+	var response *resources.Response
+	if mapper != nil {
+		response = mapper.MapError(err)
+	}
+
+	if response == nil {
+		response = mapErrorResponse(err)
 	}
+	return response
+}
+
+// writeServiceErrorResponse is a helper method that maps errors thrown by
+// services and writes them to the HTTP response stream.
+func writeServiceErrorResponse(w http.ResponseWriter, mapper ErrorResponseMapper, err error) {
```

Comment:
honestly, I'm not a fan of this since Mapping and Writing IMHO should not live under the same roof, but I'll step aside on this one

Reply:
I know what you say, but note that this is a helper method to avoid repeating ourselves, and of course, ensure we're not missing the steps.